### PR TITLE
Ew new button config (CU-86dte4ejq)

### PR DESCRIPTION
**Pull request by elanawood**
Configured BraveButtons for RAK7201v2

Test Plan:

- [x] Existing Tests Pass
- [x] Hardwear Testing: Alerts sent when pressing button '3' or '4' on the v2 buttons
- [x] Integration Testing: Added integration tests for the v2 buttons
- [x] Heartbeat: Seen on the logs and the dashboard

### CODE REVIEW

**Comment by elanawood** [comment]
I am having some trouble with the integration tests that I wrote and am going to continue to try and figure them out tomorrow!

**Comment by elanawood** [comment]
It would appear that my setup is bad :( 

Going to try to figure that out!

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ the code was deployed.
 
 ## [Unreleased]
 
+## [13.9.0] - 2024-05-08
+
+### Added
+
+- Configured buttons for RAK7201v2, now version 2 buttons will send out alerts.
```
Comment (johnny-brave) [inline CHANGELOG.md:18]: Please include the Clickup task ID for the task associated with this PR in the same format as the other entries in the Changelog.
image

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ the code was deployed.
 
 ## [Unreleased]
 
+## [13.9.0] - 2024-05-08
```
Comment (johnny-brave) [inline CHANGELOG.md:14]: This does not need to be added until we deploy to production. The changes should remain under [Unreleased] and all changes, including work done for other tasks will be moved under the new version when we deploy.

File: CHANGELOG.md [hunk 3]
```diff
@@ -724,6 +730,7 @@ the code was deployed.
 - Initial pi.
 
 [unreleased]: https://github.com/bravetechnologycoop/BraveButtons/compare/v13.8.0...HEAD
+[13.9.0]: https://github.com/bravetechnologycoop/BraveButtons/compare/v13.7.1...v13.9.0
```
Comment (johnny-brave) [inline CHANGELOG.md:733]: Similarly, this does not need to be changed until we deploy to production

File: server/rak.js [hunk 4]
```diff
@@ -13,7 +13,13 @@ const EVENT_TYPE = {
   BUTTON_PRESS_2: 66, // ASCII for 'B'
   BUTTON_PRESS_3: 67, // ASCII for 'C'
   BUTTON_PRESS_4: 68, // ASCII for 'D'
+
   HEARTBEAT: 72, // ASCII for 'H'
+
+  BUTTON_PRESS_5: 49, // ASCII for 1
+  BUTTON_PRESS_6: 50, // ASCII for 2
+  BUTTON_PRESS_7: 51, // ASCII for 3
+  BUTTON_PRESS_8: 52, // ASCII for 4
```
Comment (johnny-brave) [inline server/rak.js:22]: hmm I think these should still be named `BUTTON_PRESS_1 ... BUTTON_PRESS_4` for clarity, and then the v1 buttons should be renamed to `BUTTON_PRESS_A ... BUTTON_PRESS_D`. What do you think?

File: server/test/integration/rakTest/handleButtonPressTest.js [hunk 5]
```diff
@@ -396,4 +396,70 @@ describe('rak.js integration tests: handleButtonPress', () => {
       expect(buttonAlerts.handleValidRequest).not.to.be.called
     })
   })
+
+  describe('POST Aw== (Button 3 V2) with secondary API key for existing Button', () => {
```
Comment (johnny-brave) [inline server/test/integration/rakTest/handleButtonPressTest.js:400]: I am getting 'Mw==' for 3 encoded into base64

File: server/test/integration/rakTest/handleButtonPressTest.js [hunk 6]
```diff
@@ -396,4 +396,70 @@ describe('rak.js integration tests: handleButtonPress', () => {
...
+    it('should not log any errors', () => {
+      expect(helpers.logError).not.to.be.called
+    })
+
+    it('should handle the button press', () => {
+      expect(buttonAlerts.handleValidRequest).to.be.calledWithExactly(this.button)
+    })
+  })
+
+  describe('POST BA== (Button 4 V2) with secondary API key for existing Button', () => {
```
Comment (johnny-brave) [inline server/test/integration/rakTest/handleButtonPressTest.js:433]: I am getting 'NA==' for 4 encoded into base64

**Review by johnny-brave — CHANGES_REQUESTED** [review]
Overall, the functionality looks good! 

Thanks for updating the Changelog! Jumping the gun a little bit on the version number updates, we usually decide on the version number during deployment as it is possible that there are enough changes to warrant an upgrade in major version. 

Can you double-check on the payloads for buttons 3 and 4? That might be why the integration tests are failing.

Also, were you able to verify that the new button sends battery level information correctly? The heartbeat payload should be 'H' followed by an ASCII value that correlates to a decimal number between 0 and 100 which is the battery level

File: server/test/integration/rakTest/handleButtonPressTest.js (hunk 6 above)
Comment (elanawood) [reply server/test/integration/rakTest/handleButtonPressTest.js]: Ok I don't know how the base 64 mistakes happened I think that I converted them correctly and then wrote down the wrong thing :( but the payloads are correct now!

File: server/test/integration/rakTest/handleButtonPressTest.js (hunk 5 above)
Comment (elanawood) [reply server/test/integration/rakTest/handleButtonPressTest.js]: Fixed this!

File: server/rak.js (hunk 4 above)
Comment (elanawood) [reply server/rak.js]: Yes I think that makes sense! Made this change.

File: CHANGELOG.md (hunk 2 above)
Comment (elanawood) [reply CHANGELOG.md]: Ok! Sorry about that, I removed what I had and then left it as [xx.x.x] for when we deploy.

File: CHANGELOG.md (hunk 3 above)
Comment (elanawood) [reply CHANGELOG.md]: Fixed this, I left it in the same format as above. Let me know if that isn't ok.

File: CHANGELOG.md (hunk 1 above)
Comment (elanawood) [reply CHANGELOG.md]: Ok I added this!

File: CHANGELOG.md [hunk 7]
```diff
@@ -724,6 +730,7 @@ the code was deployed.
 - Initial pi.
 
 [unreleased]: https://github.com/bravetechnologycoop/BraveButtons/compare/v13.8.0...HEAD
+[XX.X.X]: https://github.com/bravetechnologycoop/BraveButtons/compare/v13.7.1...vXX.X.X
```
Comment (johnny-brave) [inline CHANGELOG.md:733]: I think you can just remove this line

File: CHANGELOG.md [hunk 8]
```diff
@@ -11,6 +11,12 @@ the code was deployed.
 
 ## [Unreleased]
 
+## [XX.X.X] - 2024-XX-XX
```
Comment (johnny-brave) [inline CHANGELOG.md:14]: no need for this, `## [Unreleased]` is sufficient as a header

File: server/test/unit/buttonAlertsTest/handleValidRequestTest.js [hunk 9]
```diff
@@ -627,6 +627,9 @@ describe('buttonAlerts.js unit tests: handleValidRequest', () => {
       it('should not send an alert session update', async () => {
         expect(this.sendAlertSessionUpdateStub).not.to.be.called
       })
+
+     
+
```
Comment (johnny-brave) [inline server/test/unit/buttonAlertsTest/handleValidRequestTest.js:632]: unnecessary whitespace?

File: server/rak.js [hunk 10]
```diff
@@ -9,11 +9,20 @@ const buttonAlerts = require('./buttonAlerts')
...
+  BUTTON_PRESS_C: 67, // ASCII for 'C'
+  BUTTON_PRESS_D: 68, // ASCII for 'D'
+
   HEARTBEAT: 72, // ASCII for 'H'
+
+
+  BUTTON_PRESS_1: 49, // ASCII for 1
+  BUTTON_PRESS_2: 50, // ASCII for 2
+  BUTTON_PRESS_3: 51, // ASCII for 3
+  BUTTON_PRESS_4: 52, // ASCII for 4
```
Comment (johnny-brave) [inline server/rak.js:25]: looks good! there is some whitespace that doesn't need to be there but the linter should format that for you as well.

File: CHANGELOG.md [hunk 11]
```diff
@@ -11,6 +11,12 @@ the code was deployed.
...
+
+### Added
+
+- Configured buttons for RAK7201v2, now version 2 buttons will send out alerts (CU-86dte4ejq).
```
Comment (johnny-brave) [inline CHANGELOG.md:18]: perfect!

**Review by johnny-brave — CHANGES_REQUESTED** [review]
Looks good! Once you get the tests passing, we'll be ready to merge.

File: CHANGELOG.md (hunk 11 above)
Comment (elanawood) [reply CHANGELOG.md:18]: Yay!

File: server/rak.js (hunk 10 above)
Comment (elanawood) [reply server/rak.js]: Fixed this!

File: server/test/unit/buttonAlertsTest/handleValidRequestTest.js (hunk 9 above)
Comment (elanawood) [reply server/test/unit/buttonAlertsTest/handleValidRequestTest.js]: Fixed this, I'm not sure why that happened.

File: CHANGELOG.md (hunk 8 above)
Comment (elanawood) [reply CHANGELOG.md]: Fixed!

File: CHANGELOG.md (hunk 7 above)
Comment (elanawood) [reply CHANGELOG.md]: Fixed!

File: server/rak.js [hunk 12]
```diff
@@ -9,11 +9,15 @@ const buttonAlerts = require('./buttonAlerts')
 const rakApiKeys = [helpers.getEnvVar('RAK_API_KEY_PRIMARY'), helpers.getEnvVar('RAK_API_KEY_SECONDARY')]
 
 const EVENT_TYPE = {
-  BUTTON_PRESS_1: 65, // ASCII for 'A'
-  BUTTON_PRESS_2: 66, // ASCII for 'B'
-  BUTTON_PRESS_3: 67, // ASCII for 'C'
-  BUTTON_PRESS_4: 68, // ASCII for 'D'
+  BUTTON_PRESS_A: 65, // ASCII for 'A'
```
Comment (micahdbak) [inline server/rak.js:12]: Maybe re-order these so the ASCII values are ordered; the ordering would then be like:

BUTTON_PRESS_1,
BUTTON_PRESS_2,
BUTTON_PRESS_3,
BUTTON_PRESS_4,
BUTTON_PRESS_A,
BUTTON_PRESS_B,
BUTTON_PRESS_C,
BUTTON_PRESS_D,
HEARTBEAT

Comment (micahdbak) [reply server/rak.js:12]: Also good that you renamed the old values so the BUTTON_PRESS_X corresponds to what ASCII character it references!

File: server/rak.js [hunk 13]
```diff
@@ -40,7 +44,12 @@ async function handleButtonPress(req, res) {
 
       if (event[0] === EVENT_TYPE.HEARTBEAT && button !== null) {
         await db.logButtonsVital(button.id, event[1], snr, rssi)
-      } else if (event[0] === EVENT_TYPE.BUTTON_PRESS_4 || event[0] === EVENT_TYPE.BUTTON_PRESS_3) {
+      } else if (
+        event[0] === EVENT_TYPE.BUTTON_PRESS_D ||
+        event[0] === EVENT_TYPE.BUTTON_PRESS_C ||
+        event[0] === EVENT_TYPE.BUTTON_PRESS_4 ||
+        event[0] === EVENT_TYPE.BUTTON_PRESS_3
```
Comment (micahdbak) [inline server/rak.js:51]: Change the ordering of these checks as well; 3, 4, C, D makes the most sense to me

File: server/test/integration/rakTest/handleButtonPressTest.js [hunk 14]
```diff
@@ -396,4 +396,70 @@ describe('rak.js integration tests: handleButtonPress', () => {
...
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.logError).not.to.be.called
+    })
+
+    it('should handle the button press', () => {
+      expect(buttonAlerts.handleValidRequest).to.be.calledWithExactly(this.button)
+    })
+  })
```
Comment (micahdbak) [inline server/test/integration/rakTest/handleButtonPressTest.js:464]: Please add two more test cases for when the buttons 1 and 2 are pressed; these should not call handleValidRequest.

**Review by micahdbak — CHANGES_REQUESTED** [review]
Some minor changes with regards to ordering and exhaustive test cases; overall everything looks good! I tentatively approve this PR, given you respond to my comments :)

**Comment by trisapeace** [comment]
Task linked: [CU-86dte4ejq Configure BraveButtons for RAK7201v2](https://app.clickup.com/t/86dte4ejq)

File: server/rak.js (hunk 12 above)
Comment (elanawood) [reply server/rak.js:12]: Woooo, that was Johnny's suggestion!

File: server/test/integration/rakTest/handleButtonPressTest.js (hunk 14 above)
Comment (elanawood) [reply server/test/integration/rakTest/handleButtonPressTest.js:464]: Done!

File: server/rak.js (hunk 13 above)
Comment (elanawood) [reply server/rak.js]: Done!

**Comment by elanawood** [comment]
Made all of the changes & I am pretty sure this is ready to merge!

**Review by micahdbak — APPROVED** [review]
Looks good; well done!

**Comment by micahdbak** [comment]
Oh yeah, please make sure to Squash & Merge when you're ready!
